framework/wazuh/tests/test_agent.py

```diff
@@ -84,6 +84,69 @@ def test_agent_get_distinct_agents(socket_mock, send_mock, fields, expected_item
     assert distinct.affected_items == expected_items, f'"Affected_items" does not match. Should be "{expected_items}".'
 
 
+@pytest.mark.parametrize('agent_list, fields, order, expected_items', [
+    (['000', '002'],
+     ['os.name', 'os.version'],
+     ['asc'],
+     [
+        {'id': '002', 'os': {'name': 'Ubuntu', 'version': '16.04.1 LTS'}},
+        {'id': '000', 'os': {'name': 'Ubuntu', 'version': '18.04.1 LTS'}}
+     ]
+     ),
+    (['000', '002'],
+     ['os.name', 'os.version'],
+     ['desc'],
+     [
+        {'id': '000', 'os': {'name': 'Ubuntu', 'version': '18.04.1 LTS'}},
+        {'id': '002', 'os': {'name': 'Ubuntu', 'version': '16.04.1 LTS'}}
+     ]
+     ),
+    (['000', '002', '009'],
+     ['os.version', 'os.name'],
+     ['asc'],
+     [
+         {'id': '009', 'os': {'name': 'Windows', 'version': '3.14 XP'}},
+         {'id': '002', 'os': {'name': 'Ubuntu', 'version': '16.04.1 LTS'}},
+         {'id': '000', 'os': {'name': 'Ubuntu', 'version': '18.04.1 LTS'}}
+     ]
+     ),
+    (['000', '002', '009'],
+     ['os.version', 'os.name'],
+     ['desc'],
+     [
+         {'id': '000', 'os': {'name': 'Ubuntu', 'version': '18.04.1 LTS'}},
+         {'id': '002', 'os': {'name': 'Ubuntu', 'version': '16.04.1 LTS'}},
+         {'id': '009', 'os': {'name': 'Windows', 'version': '3.14 XP'}},
+     ]
+     ),
+    (['000', '002', '009'],
+     ['os.name', 'os.version'],
+     ['asc'],
+     [
+         {'id': '002', 'os': {'name': 'Ubuntu', 'version': '16.04.1 LTS'}},
+         {'id': '000', 'os': {'name': 'Ubuntu', 'version': '18.04.1 LTS'}},
+         {'id': '009', 'os': {'name': 'Windows', 'version': '3.14 XP'}},
+     ]
+     ),
+])
+@patch('wazuh.core.common.client_keys', new=os.path.join(test_agent_path, 'client.keys'))
+@patch('wazuh.core.wdb.WazuhDBConnection._send', side_effect=send_msg_to_wdb)
+@patch('socket.socket.connect')
+def test_agent_sort_order(socket_mock, send_mock, agent_list, order, fields, expected_items):
+    """Test `sort` parameter of GET /agents endpoint
+
+    Parameters
+    ----------
+    fields : list
+        List of fields to check their values.
+    expected_items : list
+        List of expected values for the provided fields.
+    """
+    distinct = get_agents(agent_list=agent_list, select=fields, sort={'fields': fields, 'order': order[0]})
```

Comment:
Why is order a list?? order could be a string and you will avoid using order[0].
Also change variable name as it may be confusing.